Decisiv/styled-components-modifiers
# Offers suggestions for improved typings and build processes

## OVERVIEW
_Give a brief description of what this PR does._

This PR offers suggested feedback to #24. Specifically, some issues around types were found when linking this package to a project using it. There are also some new package configurations that are updated.

## WHERE SHOULD THE REVIEWER START?
package.json

## HOW CAN THIS BE MANUALLY TESTED?
Run the test suite.
Build the package.
Link the package locally to a project being built in TypeScript and ensure the types match up as expected.

## ANY NEW DEPENDENCIES ADDED?
Several added, updated, and moved around. Review the package.json

## CHECKLIST
_Be sure all items are_ ✅ _before submitting a PR for review._
* [ ] Verify the linter and tests pass: `npm run review`
* [ ] Verify this branch is rebased with the latest master

## GIF
_Share a fun GIF to say thanks to your reviewer:_
https://giphy.com

![](https://media.giphy.com/media/xTiTnfkt9wCx4fuWhW/giphy.gif)
